pkg/util/cgroups/cgroupv1_cpu.go

```diff
@@ -42,20 +43,21 @@ func (c *cgroupV1) parseCPUController(stats *CPUStats) {
 		reportError(err)
 	}
 
-	if err := parse2ColumnStats(c.fr, c.pathFor("cpu", "cpu.stat"), 0, 1, func(key, value string) error {
-		intVal, err := strconv.ParseUint(value, 10, 64)
+	if err := parse2ColumnStats(c.fr, c.pathFor("cpu", "cpu.stat"), 0, 1, func(key, value []byte) error {
+		intVal, err := strconv.ParseUint(string(value), 10, 64)
 		if err != nil {
-			reportError(newValueError(value, err))
+			reportError(newValueError(string(value), err))
 			// Dont't stop parsing on a single faulty value
 			return nil
 		}
 
-		switch key {
-		case "nr_throttled":
+		if bytes.Equal(key, []byte("nr_throttled")) {
```

Comment:
This can be converted back to a switch; profiling shows that the compiler is smart enough to not allocate

Reply:
could you convert back to switch 🙇